SU2_PY/SU2_Nastran/pysu2_nastran.py

```diff
@@ -0,0 +1,912 @@
+
+    self.q_n = np.copy(self.q)
+    self.qdot_n = np.copy(self.qdot)
+    self.qddot_n = np.copy(self.qddot)
+    self.a_n = np.copy(self.a)
+    self.__reset(self.q)
+    self.__reset(self.qdot)
+    self.__reset(self.qddot)
+    self.__reset(self.a)
+
+    makerID = list(self.markers.keys())
+    makerID = makerID[0]
+    nodeList = self.markers[makerID]
+
+    for iPoint in nodeList:
+      self.node[iPoint].updateCoordVel()
+
+
+  def applyload(self, iVertex, fx, fy, fz):
+    """ Description """
+
+    makerID = list(self.markers.keys())
+    makerID = makerID[0]
+    iPoint = self.getInterfaceNodeGlobalIndex(makerID, iVertex)
+    self.node[iPoint].SetForce((fx,fy,fz))
+
+  def getFSIMarkerID(self):
+    """ Description. """
+    L = list(self.markers)
+    return L[0]
+
+  def getNumberOfSolidInterfaceNodes(self, markerID):
+    """ Description. """
+
+    return len(self.markers[markerID])
+
+  def getInterfaceNodeGlobalIndex(self, markerID, iVertex):
+    """ Description. """
+
+    return self.markers[markerID][iVertex]
+
+  def getInterfaceNodePosX(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    return float(Coord[0])
+
+  def getInterfaceNodePosY(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    return float(Coord[1])
+
+  def getInterfaceNodePosZ(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    return float(Coord[2])
+
+  def getInterfaceNodeDispX(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    Coord0 = self.node[iPoint].GetCoord0()
+    return float(Coord[0]-Coord0[0])
+
+  def getInterfaceNodeDispY(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    Coord0 = self.node[iPoint].GetCoord0()
+    return float(Coord[1]-Coord0[1])
+
+  def getInterfaceNodeDispZ(self, markerID, iVertex):
+    """ Desciption. """
+
+    iPoint = self.markers[markerID][iVertex]
+    Coord = self.node[iPoint].GetCoord()
+    Coord0 = self.node[iPoint].GetCoord0()
+    return float(Coord[2]-Coord0[2])
+
+  def getInterfaceNodeVelX(self, markerID, iVertex):
+    """ Description """
+
+    iPoint = self.markers[markerID][iVertex]
+    Vel = self.node[iPoint].GetVel()
+    return float(Vel[0])
+
+  def getInterfaceNodeVelY(self, markerID, iVertex):
+    """ Description """
+
+    iPoint = self.markers[markerID][iVertex]
+    Vel = self.node[iPoint].GetVel()
+    return float(Vel[1])
+
+  def getInterfaceNodeVelZ(self, markerID, iVertex):
+    """ Description """
+
+    iPoint = self.markers[markerID][iVertex]
+    Vel = self.node[iPoint].GetVel()
+    return float(Vel[2])
```

Comment:
I would simplify this type of function to return an array with all the components. The caller can then access xyz if it wants.